runtime/README.md

```diff
@@ -692,107 +787,631 @@ Command Code: `0x434B_4558` ("CKEX")
+| data size      | u32      |                                   |
+| data           | u8[...]  | Data to hash                      |
+
+*Table: `CM_SHA_INIT` output arguments*
+| **Name**     | **Type** | **Description**                            |
+| ------------ | -------- | ------------------------------------------ |
+| chksum       | u32      |                                            |
+| fips_status  | u32      | FIPS approved or an error                  |
+| context size | u32      |                                            |
+| context      | u8[...]  | Passed to `CM_SHA_UPDATE` / `CM_SHA_FINAL` |
+
+*Table: `CM_SHA_INIT` / `CM_SHA_UPDATE` / `CM_SHA_FINAL` internal context*
+| **Name**          | **Type** | **Description** |
+| ----------------- | -------- | --------------- |
+| input buffer      | u8[128]  |                 |
+| intermediate hash | u8[64]   |                 |
+| length            | u64      |                 |
+| hash algorithm    | u32      |                 |
+
+### CM_SHA_UPDATE
+
+This continues a SHA computation started by `CM_SHA_INIT` or from another `CM_SHA_UPDATE`.
+
+The context MUST be passed in from `CM_SHA_INIT` or `CM_SHA_UPDATE`.
+
+Command Code: `0x434D_5355` ("CMSU")
+
+*Table: `CM_SHA_UPDATE` input arguments*
+| **Name**       | **Type** | **Description**                      |
+| -------------- | -------- | ------------------------------------ |
+| chksum         | u32      |                                      |
+| context size   | u32      | Size of the context.                 |
+| context        | u8[...]  | From `CM_SHA_INIT` / `CM_SHA_UPDATE` |
+| data size      | u32      |                                      |
+| data           | u8[...]  | Data to hash                         |
+
+*Table: `CM_SHA_UPDATE` output arguments*
+| **Name**     | **Type** | **Description**                            |
+| ------------ | -------- | ------------------------------------------ |
+| chksum       | u32      |                                            |
+| fips_status  | u32      | FIPS approved or an error                  |
+| context size | u32      |                                            |
+| context      | u8[...]  | Passed to `CM_SHA_UPDATE` / `CM_SHA_FINAL` |
+
+### CM_SHA_FINAL
+
+This finalizes the computation of a SHA and produces the hash of all of the data.
+
+The context MUST be passed in from `CM_SHA_INIT` or `CMA_SHA_UPDATE`.
+
+Command Code: `0x434D_5346` ("CMSF")
+
+*Table: `CM_SHA_FINAL` input arguments*
+| **Name**       | **Type** | **Description**                      |
+| -------------- | -------- | ------------------------------------ |
+| chksum         | u32      |                                      |
+| context size   | u32      | Size of the context.                 |
+| context        | u8[...]  | From `CM_SHA_INIT` / `CM_SHA_UPDATE` |
+| data size      | u32      | May be 0                             |
```

Comment:
We could. But then we get back to where we before. Because if we have an empty data, then we could really skip to the update (since the init wouldn't actually be doing anything other than calculating the starting value). And then we might as well combine the final and update steps. :)